include/crocoddyl_msgs/inertial_parameters_publisher.h

```diff
@@ -0,0 +1,74 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2023-2024, Heriot-Watt University
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_MSG_MULTIBODY_INERTIAL_PARAMETERS_PUBLISHER_H_
+#define CROCODDYL_MSG_MULTIBODY_INERTIAL_PARAMETERS_PUBLISHER_H_
+
+#include <realtime_tools/realtime_publisher.h>
+
+#include <ros/node_handle.h>
+#include "crocoddyl_msgs/MultibodyInertialParameters.h"
+
+typedef crocoddyl_msgs::MultibodyInertialParameters MultibodyInertialParameters;
+
+namespace crocoddyl_msgs {
+
+class MultibodyInertialParametersRosPublisher {
+public:
+  /**
+   * @brief Initialize the multi-body inertial parameters publisher.
+   *
+   * @param[in] n_bodies  Number of bodies
+   * @param[in] topic     Topic name
+   */
+  MultibodyInertialParametersRosPublisher(
+      const unsigned int n_bodies,
+      const std::string &topic = "/robot/multibody_inertial_parameters"):
+      n_bodies_(n_bodies) {
+    ros::NodeHandle n;
+    pub_.init(n, topic, 1);
+    pub_.msg_.parameters.resize(n_bodies_);
+  }
+
+  ~MultibodyInertialParametersRosPublisher() = default;
+
+  /**
+   * @brief Publish a multi-body inertial parameters ROS message.
+   *
+   * @param parameters[in]    multibody inertial parameters
+   */
+  void
+  publish(const std::map<std::string, Eigen::Ref<const Eigen::VectorXd> > &parameters
+          ) {
+    if (parameters.size() != n_bodies_)
+      throw std::invalid_argument(
+        "Dimension of parameters does not match n_bodies_.");
+    if (pub_.trylock()) {
+      pub_.msg_.header.stamp = ros::Time::now();
+      unsigned int i = 0;
+      for (const auto& [body_name, psi] : parameters){
+        if (psi.size() != 10)
+          throw std::invalid_argument(
+            "Dimension of psi for body " + body_name + " is not 10");
+        pub_.msg_.parameters[i].body_name = body_name;
+        for (std::size_t j=0; j<10; ++j)
+          pub_.msg_.parameters[i].psi[j] = psi[j];
+        i++;
```

Comment:
```suggestion
      for (const auto& [name, p] : parameters) {
        crocoddyl_msgs::toMsg(pub_.msg_, name, p);
      }
```

Below, I asked you to write `fromMsg` function. Please consider that, in this function, you need to support ROS 2 as well. See these lines for inspiration: https://github.com/RobotMotorIntelligence/crocoddyl_msgs/blob/devel/include/crocoddyl_msgs/conversions.h#L215-L219.

Reply:
For now I will not do as I think its unnecessary complicated